src/vector.rs

```diff
@@ -0,0 +1,69 @@
+//! Drawing [`Vector`] shapes.
+
+use bevy_ecs::prelude::*;
+use bevy_math::DVec2;
+use bevy_vello::vello::{
+    self,
+    kurbo::{self, BezPath, Shape},
+};
+
+use crate::{Fill, SceneHolder, Stroke, VelloBezPath};
+
+/// Draw [`Vector`] shapes.
+#[allow(clippy::type_complexity)]
+pub(super) fn draw_vectors<V: Vector + Component>(
+    mut commands: Commands,
+    q_vectors: Query<
+        (Entity, &V, Option<&Fill>, Option<&Stroke>),
+        Or<(Changed<V>, Changed<Fill>, Changed<Stroke>)>,
+    >,
+) {
+    for (entity, vector, fill, stroke) in q_vectors.iter() {
+        let mut scene = vello::Scene::new();
+
+        if let Some(fill) = fill {
+            scene.fill(
+                fill.style,
+                kurbo::Affine::IDENTITY,
+                &fill.brush.value,
+                Some(fill.brush.transform),
+                &vector.shape(),
+            );
+        }
+
+        if let Some(stroke) = stroke {
+            scene.stroke(
+                &stroke.style,
+                kurbo::Affine::IDENTITY,
+                &stroke.brush.value,
+                Some(stroke.brush.transform),
+                &vector.shape(),
+            );
+        }
+
+        commands
+            .entity(entity)
+            .insert(SceneHolder::<VectorScene>::new(scene));
+    }
+}
+
+/// Marker struct of a vector scene for [`SceneHolder`].
+pub struct VectorScene;
+
+pub trait Vector {
+    /// Returns vector graphics that implements [`kurbo::Shape`].
+    fn shape(&self) -> impl kurbo::Shape;
+    /// Translation of the border at a specific `time` value.
+    fn border_translation(&self, time: f64) -> DVec2 {
```

Comment:
really most (even barely complex) shapes should just keep the default impl unless we want to optimise for specific shapes (which is prob what will be done anyways) but we need bezier curve interpolation first (see other comment)